SwiftUIBestPractices.md

```diff
@@ -0,0 +1,143 @@
+struct SettingsView: View {
+    // …
+    @AppStorage(AppStorageKeys.keyTwo) private var propertyTwo = 0.75
+    @AppStorage(AppStorageKeys.keyThree) private var propertyThree = 0.5
+    @StateObject private var object = DeviceObserver()
+    
+    private let numberFormatter = NumberFormatter.fractionFormatter
+    private let wasSessionPreviouslyRunning: Bool
+    
+    // MARK: - View
+    
+    var body: some View {
+        List {...}
+            .listStyle(.insetGrouped)
+    }
+}
+```
+
+### Property Wrappers grouped together
+`@StateObject`, `@State`, and `@AppStorage` properties are marked as private, because they do not need to be mutated outside of their containing file. Property wrappers of the same type should be grouped together. 
+In the example below property wrappers are listed first followed by properties without property wrappers. 
+
+```swift 
+struct SettingsView: View {
+    @Binding private(set) var isDisplayingCertificateView: Bool
+    @ObservableObject private(set) var settingsViewModel: SettingViewModel
+    @AppStorage(AppStorageKeys.keyOne) private var propertyOne = 0.5
+    @AppStorage(AppStorageKeys.keyTwo) private var propertyTwo = 0.75
+    @AppStorage(AppStorageKeys.keyThree) private var propertyThree = 0.5
+    @StateObject private var object = DeviceObserver()
+    
+    private let numberFormatter = NumberFormatter.fractionFormatter
+    private let wasSessionPreviouslyRunning: Bool
+}
+```
+
+# Initializing SwiftUI Views
+### A synthesized initializer
+In the example below we use a synthesized initializer to create our view. We pass our binding with in our initializer to clean up our property call sites.
+
+```swift 
+struct SettingsView: View {
+    @Binding private var sliderValue: Double
+    @Binding private var wasSessionPreviouslyRunning: Bool
+    
+    private let settingsViewModel: SettingsViewModel
+    private let numberFormatter = NumberFormatter.fractionFormatter
+    
+    init(sliderValue: Binding<Double>, wasSessionPreviouslyRunning: Binding<Bool>, settingsViewModel: SettingsViewModel) {
+        self._sliderValue = sliderValue
+        self._wasSessionPreviouslyRunning = wasSessionPreviouslyRunning
+        self.settingsViewModel = settingsViewModel
+    }
+}
+```
+
+### A member-wise initializer
+If a `@Binding`  property is not being modifier with in the view which references it, consider marking it a `private(set)` . This ensures that our property won’t be mutated outside of the context we want it to be.
+
+#### Example One
+```swift 
+struct SettingsView: View {
+    
+    @Binding private(set) var sliderValue: Double
+    @Binding private(set) var wasSessionPreviouslyRunning: Bool
+    
+    private let settingsViewModel: SettingsViewModel
+    private let numberFormatter = NumberFormatter.fractionFormatter
+}
+```
+
+In this example we the values of the `@Binding` value is able to be mutated outside of this view, hence we don’t mark it `private(set)`.
```

Comment:
I’m not sure I understand this section. We aren’t modifying the binding outside of the view. Do we typically run into scenarios where bindings are exposed and read/written to elsewhere? More often, I pass bindings from one view to another, for example, but that doesn’t imply anything about access control, and each view that holds a reference could be fully `private`. I would think with SwiftUI we could reasonably expect them to always be `private`, `private(set)`, or `fileprivate(set)` depending on whether you‘re declaring multiple views in the same file, but I still am trying to understand use cases. If you could provide examples of how other types use these exposed bindings (in the comments) that might help here and we can refine this section.